cpp/ql/src/semmle/code/cpp/ir/implementation/MemoryAccessKind.qll

```diff
@@ -43,6 +80,25 @@ class PhiMemoryAccess extends MemoryAccessKind, TPhiMemoryAccess {
   }
 }
 
+/**
+ * The operand is a ChiOld operand, which accesses the same memory as its
+ * definition.
+ */
+class ChiTotalMemoryAccess extends MemoryAccessKind, TChiOldMemoryAccess {
+  override string toString() {
+    result = "chi(total)"
+  }
+}
+
+/**
+ * The operand is a ChiUpdate operand, which accesses the same memory as its
+ * definition.
```

Comment:
This comment, and the one for `ChiTotalMemoryAccess`, don't really tell me anything. They refer to something called "ChiOld" and "ChiUpdate", but what does that mean?